core/src/main/java/org/apache/iceberg/DataFilesTable.java

```diff
@@ -144,8 +144,9 @@ private ManifestReadTask(FileIO io, FileScanTask manifestTask, Schema schema) {
 
     @Override
     public CloseableIterable<StructLike> rows() {
+      String manifestPath = manifestTask.file().path().toString();
       return CloseableIterable.transform(
-          ManifestReader.read(io.newInputFile(manifestTask.file().path().toString())).project(schema),
+          ManifestReader.readEntries(io.newInputFile(manifestPath)).project(schema),
```

Comment:
What is the purpose of using `readEntries` here? Just to avoid assigning inheritable metadata that will be ignored?

In general, I'm wondering if we need `readEntries` because I don't want library maintainers to accidentally get manifest entries without inherited metadata and accidentally change the inherited values. It would be easier and more reliable to always return entries with inherited metadata.

Reply:
I've updated metadata tables to get rid of `readEntries`.